evennia/contrib/rpg/rpsystem/rpsystem.py

```diff
@@ -619,66 +534,41 @@ def send_emote(sender, receivers, emote, anonymous_add="first", **kwargs):
     # if anonymous_add is passed as a kwarg, collect and remove it from kwargs
     if "anonymous_add" in kwargs:
         anonymous_add = kwargs.pop("anonymous_add")
-    if anonymous_add and not any(1 for tag in obj_mapping if tag.startswith(skey)):
+    self_refs = [f"{skey}{ref}" for ref in ('t','^','v','~','')]
+    if anonymous_add and not any(1 for tag in obj_mapping if tag in self_refs):
         # no self-reference in the emote - add to the end
-        obj_mapping[skey] = sender
         if anonymous_add == "first":
+            skey = skey + 't'
             possessive = "" if emote.startswith("'") else " "
             emote = "%s%s%s" % ("{{%s}}" % skey, possessive, emote)
         else:
             emote = "%s [%s]" % (emote, "{{%s}}" % skey)
+        obj_mapping[skey] = sender
 
     # broadcast emote to everyone
     for receiver in receivers:
         # first handle the language mapping, which always produce different keys ##nn
         receiver_lang_mapping = {}
```

Comment:
It seems this is no longer needed since `receiver_lang_mapping` is now created in-place on the lines below.

Reply:
The purpose of it is to include a mapping for the self-reference sdesc key that was just added via `anonymous_add`. I just moved it to the end of the `if` block instead of the beginning in order to account for the potentially added case ref.

Reply:
I don't see what you mean? This declaration of an empty `receiver_lang_mapping` is immediately followed by an `if...else` block where both conditions generate `receiver_lang_mapping` from scratch.

Reply:
Ohhhh, I see where I got confused, you're suggesting I take the unnecessary declaration out, not remarking on the change I made. Good catch, I'll do that.